src/cascadia/TerminalSettingsEditor/Profiles_Base.cpp

```diff
@@ -37,6 +37,25 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         {
             StartingDirectoryUseParentCheckbox().IsChecked(true);
         }
+
+        // Initialize the terminal only once the swapchainpanel is loaded - that
+        //      way, we'll be able to query the real pixel size it got on layout
+        _layoutUpdatedRevoker = LayoutUpdated(winrt::auto_revoke, [state, this](auto /*s*/, auto /*e*/) {
+            // This event fires every time the layout changes, but it is always the last one to fire
+            // in any layout change chain. That gives us great flexibility in finding the right point
+            // at which to initialize our renderer (and our terminal).
+            // Any earlier than the last layout update and we may not know the terminal's starting size.
+
+            // Only let this succeed once.
+            _layoutUpdatedRevoker.revoke();
+
+            if (state.FocusDeleteButton())
+            {
+                DeleteButton().Focus(FocusState::Programmatic);
+                state.FocusDeleteButton(false);
+                ProfilesBase_ScrollView().ChangeView(nullptr, ProfilesBase_ScrollView().ScrollableHeight(), nullptr);
```

Comment:
Instead of passing a boolean I'd pass the `VerticalOffset` of the ScrollView and restore the scroll offset accurately.

Alternatively, why don't we just set `BringIntoViewOnFocusChange="true"`? Does this not work?

Reply:
> Instead of passing a boolean I'd pass the `VerticalOffset` of the ScrollView and restore the scroll offset accurately.

Because the bug is specifically about making sure to focus the delete button, so KB users don't have to tab through all the settings



> Alternatively, why don't we just set `BringIntoViewOnFocusChange="true"`? Does this not work?

I did not know about that, lemme try that one

Reply:
Just double checked - that didn't work in this case 🤷